contracts/ibc-reflect-send/src/contract.rs

```diff
@@ -0,0 +1,649 @@
+        data: to_binary(&packet)?,
+        timeout_block: None,
+        timeout_timestamp: build_timeout(&env.block),
+    };
+
+    Ok(HandleResponse {
+        messages: vec![msg.into()],
+        attributes: vec![attr("action", "handle_send_msgs")],
+        data: None,
+    })
+}
+
+pub fn handle_check_remote_balance(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    channel_id: String,
+) -> StdResult<HandleResponse> {
+    // auth check
+    let cfg = config(deps.storage).load()?;
+    if info.sender != cfg.admin {
+        return Err(StdError::generic_err("Only admin may send messages"));
+    }
+    // ensure the channel exists (not found if not registered)
+    accounts(deps.storage).load(channel_id.as_bytes())?;
+
+    // construct a packet to send
+    let packet = PacketMsg::Balances {};
+    let msg = IbcMsg::SendPacket {
+        channel_id,
+        data: to_binary(&packet)?,
+        timeout_block: None,
+        timeout_timestamp: build_timeout(&env.block),
+    };
+
+    Ok(HandleResponse {
+        messages: vec![msg.into()],
+        attributes: vec![attr("action", "handle_check_remote_balance")],
+        data: None,
+    })
+}
+
+pub fn handle_send_funds(
+    deps: DepsMut,
+    env: Env,
+    mut info: MessageInfo,
+    reflect_channel_id: String,
+    transfer_channel_id: String,
+) -> StdResult<HandleResponse> {
+    // intentionally no auth check
+
+    // require some funds
+    if info.sent_funds.is_empty() {
+        return Err(StdError::generic_err(
+            "you must send the coins you with to ibc transfer",
+        ));
+    } else if info.sent_funds.len() > 1 {
+        return Err(StdError::generic_err("you can only ibc transfer one coin"));
+    }
+    let amount = info.sent_funds.swap_remove(0);
+
+    // load remote account
+    let data = accounts(deps.storage).load(reflect_channel_id.as_bytes())?;
+    let remote_addr = match data.remote_addr {
+        Some(addr) => addr,
+        None => {
+            return Err(StdError::generic_err(
+                "We don't have the remote address for this channel",
+            ))
+        }
+    };
+
+    // construct a packet to send
+    let msg = IbcMsg::Transfer {
+        channel_id: transfer_channel_id,
+        to_address: remote_addr,
+        amount,
+        timeout_block: None,
+        timeout_timestamp: build_timeout(&env.block),
+    };
+
+    Ok(HandleResponse {
+        messages: vec![msg.into()],
+        attributes: vec![attr("action", "handle_send_funds")],
+        data: None,
+    })
+}
+
+#[entry_point]
+pub fn query(deps: Deps, _env: Env, msg: QueryMsg) -> StdResult<QueryResponse> {
+    match msg {
+        QueryMsg::Admin {} => to_binary(&query_admin(deps)?),
+        QueryMsg::Account { channel_id } => to_binary(&query_account(deps, channel_id)?),
+        QueryMsg::ListAccounts {} => to_binary(&query_list_accounts(deps)?),
+    }
+}
+
+pub fn query_account(deps: Deps, channel_id: String) -> StdResult<AccountResponse> {
```

Comment:
Why are all of those quey implementations `pub`?

Reply:
To let me unit test easier. But I can change that (and align with the integration tests)

Reply:
The beauty of the in-file unit testing in Rust is that you by default can unit-test private functions. They are available in the `tests` sub-module. So you can keep everything as-is and remove the `pub`.